### ISSUE TITLE
Misleading / unhelpful warning tooltip on 4.0 -> 4.1 upgrade

### ISSUE THREAD

**Issue by romlok** [godotengine/godot#79095]
### Godot version

v4.1.stable.official [970459615]  < Steam

### System information

Godot v4.1.stable - Debian GNU/Linux trixie/sid trixie - Vulkan (Mobile) - dedicated NVIDIA GeForce GTX 1060 (nvidia) - Intel(R) Core(TM) i5-8400 CPU @ 2.80GHz (6 Threads)

### Issue description

After being upgraded to 4.1 via Steam (before any release announcement even?), I saw the warning marker against my game in the project manager, and noticed that the tooltip wasn't all that helpful:

![Screenshot_20230706_121443](https://github.com/godotengine/godot/assets/441188/7e62f67c-664f-4f5f-8f84-0cd0084679a6)

It seems to think that the old version number is a deprecated feature? :/

### Steps to reproduce

1. Have a project from Godot 4.0
2. Upgrade Godot to 4.1
3. Open project manager

Alternatively:
1. Create a new project in 4.1
2. Quit Godot
3. Edit the `project.godot`'s `config/features` to read `4.0` instead of `4.1`
4. Open project manager

### Minimal reproduction project

[default.zip](https://github.com/godotengine/godot/files/11967512/default.zip)

**Comment by YuriSizov** [comment]
> It seems to think that the old version number is a deprecated feature? :/

The wording is unfortunate for this specific case, but that's just the nature of how the version check is implemented. I guess we could somehow handle it as a special case.

**Comment by romlok** [comment]
Yeah, IMO the version number does seem to be a special case, separate from the explicit list of features in use.

My expectation would be that there wouldn't be any warning at all if it's only the version number which differs.   
Or else the warning would explicitly state that the project is from an earlier version of the engine, so the user is informed to be on the lookout for things which are not necessarily deprecated, but may not work quite as they used to (though the confirmation dialog when opening the project may also serve that purpose?).

**Comment by aaronfranke** [comment]
Here is my proposed fix: https://github.com/godotengine/godot/pull/79118
